viewing_party/party.py

```diff
@@ -1,23 +1,183 @@
+    else:
+        # update movie_genre_count based on watch history
+        for movie in user_data["watched"]:
+            genre = movie["genre"]
+            if genre not in movie_genre_count:
+                movie_genre_count[genre] = 1
+            else:
+                movie_genre_count[genre] += 1
+
+        # convert movie_genre_count keys and values to lists to allow for max function
+        key_list = list(movie_genre_count.keys())
+        value_list = list(movie_genre_count.values())
+        max_value = max(value_list)
+
+        # identify index of most popular genre
+        popular_genre_index = value_list.index(max_value)
+
+        # identify the most frequently watched movie genre
+        popular_genre = key_list[popular_genre_index]
+
+    return popular_genre
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+
+    # create lists to store unique movies and movies that user and friends have watched
+    movies_in_common = []
+    my_unique_movies = []
+
+    # loop through friends' watched movies to identify movies user and friends have both watched
+    for i in range(len(user_data["friends"])):
+        for movie in user_data["friends"][i]["watched"]:
+            if movie in user_data["watched"]:
+                movies_in_common.append(movie)
+    
+    # loop through user's watched movies and compare to movies in common to find unique movies
+    for my_movie in user_data["watched"]:
+        if my_movie not in movies_in_common:
+            my_unique_movies.append(my_movie)
+    
+    return my_unique_movies
+
+def get_friends_unique_watched(user_data):
+
+    # list to store movies only friends have watched
+    friends_unique_movies = []
+
+    # loop through friends' watched movies to identify movies that friends have watched, but user hasn't watched
+    for i in range(len(user_data["friends"])):
+        for movie in user_data["friends"][i]["watched"]:
+            if movie not in user_data["watched"] and movie not in friends_unique_movies:
+                friends_unique_movies.append(movie)
+    
+    return friends_unique_movies
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+
+    # empty list to hold movie recommendations
+    my_recommended_movies = []
+
+    # loop through all of the movies that the user's friends have watched
+    for i in range(len(user_data["friends"])):
+        for movie in user_data["friends"][i]["watched"]:
+            # check that the user hasn't watched the movie, the user has the movie's correct subscription service, and the movie isn't already recommended
+            if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"] and movie not in my_recommended_movies:

```

Comment:
This line is pretty long, we should break it across a couple lines to make it easier to read. [Here's a bit more info recommended styles for breaking up lines](https://realpython.com/python-pep8/#:~:text=PEP%208%20suggests%20lines%20should,to%20run%20over%20several%20lines.).